pkg/cli/cvd.go

```diff
@@ -358,24 +361,56 @@ func credentialsFactoryFromSource(source string) (CredentialsFactory, error) {
 	case InjectedCredentialsSource:
 		return func() string { return client.InjectedCredentials }, nil
 	default:
-		return jwtAccessToken(source)
+		// expected: `(jwt|oauth):/dir/credentialFile`
+		return accessToken(source)
 	}
 }
 
-func jwtAccessToken(filepath string) (func() string, error) {
+const (
+	jwtAuthType   = "jwt"
+	oauthAuthType = "oauth"
+)
+
+func accessToken(source string) (func() string, error) {
+	re := regexp.MustCompile(`^(jwt|oauth):([^\]]+)$`)
+	match := re.FindAllStringSubmatch(source, -1)
+	if match == nil || len(match) > 1 {
+		return nil, errors.New("unknown credential type, only accept: `none`/`injected`/`jwt:'<filepath>'`/`oauth:'<filepath>'`")
+	}
+	authType, filepath := match[0][1], match[0][2]
```

Comment:
Couldn't we simplify the code with using `strings.SplitN` and avoid using regex? I think we're already validating for `authType` and `filepath` below. `strings.SplitN` seems to be enough.

Reply:
Fixed. Thanks for the suggestion!